File: renderer/src/main/java/ca/on/hojat/renderer/exif/ExifUtil.java
package ca.on.hojat.renderer.exif;

import java.text.NumberFormat;
import java.text.DecimalFormat;


public class ExifUtil {

    static final NumberFormat formatter = DecimalFormat.getInstance();

    public static String processLensSpecifications(Rational[] values) {
        Rational min_focal = values[0];
        Rational max_focal = values[1];
        Rational min_f = values[2];
        Rational max_f = values[3];

        formatter.setMaximumFractionDigits(1);

        return formatter.format(min_focal.toDouble()) +
                "-" +
                formatter.format(max_focal.toDouble()) +
                "mm f/" +
                formatter.format(min_f.toDouble()) +
                "-" +
                formatter.format(max_f.toDouble());
    }

}

File: renderer/src/main/java/ca/on/hojat/renderer/exif/ExifUtil.kt
package ca.on.hojat.renderer.exif

import java.text.DecimalFormat
import java.text.NumberFormat

object ExifUtil {

    private val formatter: NumberFormat = DecimalFormat.getInstance()

    @JvmStatic
    fun processLensSpecifications(values: Array<Rational>): String {
        val minFocal = values[0]
        val maxFocal = values[1]
        val minF = values[2]
        val maxF = values[3]

        formatter.maximumFractionDigits = 1
        return "${formatter.format(minFocal.toDouble())}-${formatter.format(maxFocal.toDouble())}mm f/${
            formatter.format(
                minF.toDouble()
            )
        }-${formatter.format(maxF.toDouble())}"
    }
}